### PR TITLE
Add missed settings update consumer

### PR DESCRIPTION
**Pull request by spbjss**
### Description
Add missed settings update consumer
 
### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/40
 
### Check List
- [ Y ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java [hunk 1]
```diff
@@ -75,9 +76,10 @@
     private CheckpointDao checkpointDao;
     private CacheProvider cache;
     private final NodeStateManager stateManager;
-    private final int coolDownMinutes;
+    private int coolDownMinutes;
```
Comment (ylwu-amzn) [inline src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java:79]: Will the update consumer work by using "private int" ? I think we should use "private volatile int"

Comment (spbjss) [reply src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java:79]: Thanks! will update in another commit.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

File: src/main/java/org/opensearch/ad/transport/BackPressureRouting.java [hunk 2]
```diff
@@ -45,18 +46,23 @@
...
 
-    public BackPressureRouting(String nodeId, Clock clock, Settings settings) {
+    public BackPressureRouting(String nodeId, Clock clock, Settings settings, ClusterService clusterService) {
         this.nodeId = nodeId;
         this.clock = clock;
         this.backpressureCounter = new AtomicInteger(0);
         this.maxRetryForUnresponsiveNode = MAX_RETRY_FOR_UNRESPONSIVE_NODE.get(settings);
         this.mutePeriod = BACKOFF_MINUTES.get(settings);
         this.lastMuteTime = 0;
+        this.clusterService = clusterService;
```
Comment (kaituo) [inline src/main/java/org/opensearch/ad/transport/BackPressureRouting.java:62]: Have you tried to verify dynamically changing max retry for unresponsive node works?  You need to simulate a node is not responsive by changing the code to reject every incoming request and check if the node will be filtered out after the set unresponsive times?

Comment (kaituo) [reply src/main/java/org/opensearch/ad/transport/BackPressureRouting.java:62]: This applies to all settings you have changed.

**Pull request closed without merging**